Modules/ThirdParty/Eigen3/src/itkeigen/CMakeLists.txt

```diff
@@ -686,8 +686,15 @@ else ()
   add_library (eigen INTERFACE)
   add_library (Eigen3::Eigen ALIAS eigen)
 
-  set(EIGEN_DEFINITIONS "EIGEN_MPL2_ONLY")
-  target_compile_definitions (eigen INTERFACE ${EIGEN_DEFINITIONS})
+  if(NOT DEFINED ITK_USE_EIGEN_MPL2_ONLY)
+    option(ITK_USE_EIGEN_MPL2_ONLY "Set compile definition EIGEN_MPL2_ONLY. It might leak to when third parties use Eigen and ITK. Use only internally." OFF)
```

Comment:
It seems that `ON` is a better default so we do not inadvertently use LGPL code in ITK?

Reply:
I think it is better to have it OFF by default. If ON, then third-parties using both, ITK and Eigen3 (with LGPL headers), would require that ITK is built with a non-default option.

I would say we enable that option in the CI robots, for internal checking, what do you think?

Reply:
Ok, we could add it to the CI. Could you please create a PR for the script on the `dashboard` branch?

Reply:
Yes, will do, thanks Matt.